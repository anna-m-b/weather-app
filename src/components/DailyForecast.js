import React from 'react';
import PropTypes from 'prop-types';
import WeatherIcon from 'react-icons-weather';
import getStringyDate from '../util/getStringyDate';
import '../styles/DailyForecast.css';
import '../styles/weather-icons.min.css';

const DailyForecast = ({
  date,
  icon,
  temperature,
  description,
  handleMoreDetailsClick,
}) => {
  return (
    <div className="DailyForecast" data-testid="daily-forecast">
      <div className="DailyForecast_date">{getStringyDate(date)}</div>
      <WeatherIcon
        className="DailyForecast_icon"
        name="owm"
        iconId={icon}
        aria-label={`${description} icon`}
        data-testid="weather-icon"
      />
      <div className="DailyForecast_temperature">{`${temperature}Â°c`}</div>
      <div className="DailyForecast_description">{description}</div>
      <button
        type="button"
        onClick={() => handleMoreDetailsClick(date)}
        className="DailyForecast_more-details"
      >
        More details
      </button>
    </div>
  );
};

DailyForecast.propTypes = {
  date: PropTypes.number.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
    .isRequired,
  description: PropTypes.string.isRequired,
  handleMoreDetailsClick: PropTypes.func.isRequired,
};

export default DailyForecast;
