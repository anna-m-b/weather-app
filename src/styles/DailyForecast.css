.DailyForecast {
  margin: 5px;
  /* border: 0.5px solid blue; */
  border-radius: 15px;
  background-color: #ffffff38;
  width: 6.5rem;
  height: 9rem;
  display: grid;
  align-items: center;
  padding: 5px;
  font-size: 1rem;
  box-shadow: 3px 0px 43px -1px rgba(82,120,210,0.75);
  -webkit-box-shadow: 3px 0px 43px -1px rgba(82,120,210,0.75);
  -moz-box-shadow: 3px 0px 43px -1px rgba(82,120,210,0.75);
}

.DailyForecast_icon, .DailyForecast_date {
  justify-self: center;
}

.DailyForecast_temperature, .DailyForecast_description, .DailyForecast_more-details {
  padding-left: 0.6rem;
}

.DailyForecast_icon {
  font-size: 2.3rem;
  margin: 0.2rem;
  color: #160064;
}


.DailyForecast_date {
  font-weight: 500;
}

.DailyForecast_more-details {
  font-size: 0.6rem;
}