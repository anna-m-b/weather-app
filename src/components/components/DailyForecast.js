import React from 'react';
import '../../styles/DailyForecast.css';
import '../../styles/weather-icons.min.css';

const DailyForecast = () => {
  return (
    <div className="DailyForecast">
      <p className="date">Sat 6th Feb</p>
      <i className="wi wi-day-rain" />
      <p>
        6Â°c
        <br />
        Rainy
        <br />
        <span className="more-details">More details</span>
      </p>
    </div>
  );
};

export default DailyForecast;
