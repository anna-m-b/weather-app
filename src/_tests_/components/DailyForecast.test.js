import React from 'react';
import { render, screen } from '@testing-library/react';
import DailyForecast from '../../components/DailyForecast';

describe('DailyForecast', () => {
  const validProps = {
    date: 1525392000000,
    icon: '501',
    description: 'Rainy',
    temperature: 6,
    handleMoreDetailsClick: () => {},
  };

  it('renders correctly', () => {
    const { asFragment } = render(<DailyForecast {...validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('it renders each element based on props', () => {
    beforeEach(() => {
      render(<DailyForecast {...validProps} />);
    });

    it('renders the date', () => {
      expect(screen.getByText('Fri 4 May')).toBeTruthy();
      expect(screen.getByText('Fri 4 May')).toHaveClass('DailyForecast_date');
    });

    it('renders a weather icon', () => {
      expect(
        screen.getByTestId('weather-icon', { name: /rainy/i })
      ).toBeTruthy();
      expect(
        screen.getByTestId('weather-icon', { name: /rainy/i })
      ).toHaveClass('DailyForecast_icon');
    });

    it('renders the temperature', () => {
      expect(screen.getByText('6°c')).toBeTruthy();
      expect(screen.getByText('6°c')).toHaveClass('DailyForecast_temperature');
    });

    it('renders a weather description as text', () => {
      expect(screen.getByText('Rainy')).toBeTruthy();
      expect(screen.getByText('Rainy')).toHaveClass(
        'DailyForecast_description'
      );
    });
  });
});
