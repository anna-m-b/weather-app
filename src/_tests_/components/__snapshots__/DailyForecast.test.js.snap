// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyForecast renders correctly 1`] = `
<DocumentFragment>
  <div
    class="DailyForecast"
    data-testid="daily-forecast"
  >
    <div
      class="DailyForecast_date"
    >
      Fri, 4 May
    </div>
    <i
      aria-label="Rainy icon"
      class="wi wi-day-rain DailyForecast_icon"
      data-testid="weather-icon"
      name="owm"
    />
    <div
      class="DailyForecast_temperature"
    >
      6Â°c
    </div>
    <div
      class="DailyForecast_description"
    >
      Rainy
    </div>
    <button
      class="DailyForecast_more-details"
      type="button"
    >
      More details
    </button>
  </div>
</DocumentFragment>
`;
